planner/main_test.go

```diff
@@ -22,107 +22,96 @@ import (
 	"github.com/canonical/github-runner-operators/internal/database"
 )
 
-func TestMain_CreateFlavor(t *testing.T) {
+func TestMain_FlavorPressure(t *testing.T) {
 	/*
 		arrange: server is listening on the configured port and prepare request payload
-		act: send create flavor request
-		assert: 201 Created and flavor exists in database with expected fields
+		act: send create flavor request and get flavor pressure request
+		assert: 201 Created and 200 OK with expected pressure value
 	*/
-	baseURL, shutdown := startTestServer(t)
-	defer shutdown()
+	go main()
+	port := os.Getenv("APP_PORT")
+	waitForHTTP(t, "http://localhost:"+port+"/api/v1/flavors/", 10*time.Second)
 
 	platform := "github"
 	labels := []string{"self-hosted", "amd64"}
 	priority := 42
 	flavor := randString(10)
+	pressure := 7
+
+	testCreateFlavor(t, port, flavor, platform, labels, priority, pressure)
+	testGetFlavorPressure(t, port, flavor, pressure)
+
```

Comment:
I would prefer more a test layout like

```
resp = createFlavor(...)
assertResponseIsHttpCreated(resp)
resp = getFlavor()
assertResponseIHttpOK(resp)
assertFlavorPressureEquals(resp, expected)
```

instead of the "sub tests" approach